diff --git a/dist/index.d.mts b/dist/index.d.mts
index f2a0a985f359e7caf49e80a4c6292282b80ce2c2..9de645f84329940bca65fd530dbe094efb858e31 100644
--- a/dist/index.d.mts
+++ b/dist/index.d.mts
@@ -69,22 +69,37 @@ interface SearchConfig {
      * @default
      * 'Search'
      */
-    btnPlaceholder?: string;
+    btnPlaceholder?: string
     /**
      * @default
      * 'Search Docs'
      */
-    placeholder?: string;
+    placeholder?: string
+    /**
+     * @default
+     * 'Close search'
+     */
+    closeButtonTitle?: string
+    /**
+     * @default
+     * 'Display detailed list'
+     */
+    displayDetailsButtonTitle?: string
+    /**
+     * @default
+     * 'Reset search'
+     */
+    resetButtonTitle?: string
     /**
      * @default
      * 'No results found.'
      */
-    emptyText?: string;
+    emptyText?: string
     /**
      * @default
      * 'Total: {{searchResult}} search results.'
      */
-    heading?: string;
+    heading?: string
     /**
      * @default
      * 'to select'
diff --git a/dist/index.d.ts b/dist/index.d.ts
index f2a0a985f359e7caf49e80a4c6292282b80ce2c2..9de645f84329940bca65fd530dbe094efb858e31 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -69,22 +69,37 @@ interface SearchConfig {
      * @default
      * 'Search'
      */
-    btnPlaceholder?: string;
+    btnPlaceholder?: string
     /**
      * @default
      * 'Search Docs'
      */
-    placeholder?: string;
+    placeholder?: string
+    /**
+     * @default
+     * 'Close search'
+     */
+    closeButtonTitle?: string
+    /**
+     * @default
+     * 'Display detailed list'
+     */
+    displayDetailsButtonTitle?: string
+    /**
+     * @default
+     * 'Reset search'
+     */
+    resetButtonTitle?: string
     /**
      * @default
      * 'No results found.'
      */
-    emptyText?: string;
+    emptyText?: string
     /**
      * @default
      * 'Total: {{searchResult}} search results.'
      */
-    heading?: string;
+    heading?: string
     /**
      * @default
      * 'to select'
diff --git a/src/Search.vue b/src/Search.vue
index f2e430925b06bc508909b4fc772a72f57c038118..b54f06640f06aa8f5ff7f7e4f1d5d4da77a5081e 100644
--- a/src/Search.vue
+++ b/src/Search.vue
@@ -13,6 +13,9 @@ import type { SearchConfig } from './type'
 import { formatPagefindResult } from './search'
 import { formatShowDate } from './utils'
 
+import VPButton from 'vitepress/dist/client/theme-default/components/VPButton.vue';
+import { OverlayScrollbarsComponent } from 'overlayscrollbars-vue';
+
 // 搜索结果
 const searchResult = ref<{ route: string; meta: Record<string, any> }[]>([])
 // 配置获取
@@ -269,7 +272,7 @@ function handleToggleDetail() {
 
 <template>
   <div class="blog-search" data-pagefind-ignore="all">
-    <div class="nav-search-btn-wait" @click="searchModal = true">
+    <VPButton tag="div" theme="alt" class="nav-search-btn-wait" @click="searchModal = true">
       <span>
         <svg width="14" height="14" viewBox="0 0 20 20">
           <path
@@ -282,13 +285,13 @@ function handleToggleDetail() {
         finalSearchConfig?.btnPlaceholder || 'Search'
       }}</span>
       <span class="metaKey"> {{ metaKey }} K </span>
-    </div>
+    </VPButton>
     <ClientOnly>
       <Command.Dialog :visible="searchModal" theme="algolia">
         <template #header>
           <div class="search-bar">
             <div class="search-actions before">
-              <button class="back-button" title="Close search" @click="searchModal = false">
+              <button class="back-button" :title="finalSearchConfig?.closeButtonTitle || 'Close search'" @click="searchModal = false">
                 <span class="vpi-arrow-left local-search-icon" />
               </button>
             </div>
@@ -299,12 +302,12 @@ function handleToggleDetail() {
             <div class="search-actions">
               <button
                 :class="{ active: showDetail }" class="toggle-layout-button" type="button"
-                title="Display detailed list" @click="handleToggleDetail"
+                :title=" finalSearchConfig?.displayDetailsButtonTitle || 'Display detailed list'" @click="handleToggleDetail"
               >
                 <span class="vpi-layout-list local-search-icon" />
               </button>
               <button
-                :disabled="!searchWords" class="clear-button" type="reset" title="Reset search"
+                :disabled="!searchWords" class="clear-button" type="reset" :title=" finalSearchConfig?.resetButtonTitle || 'Reset search' "
                 @click="handleClearSearch"
               >
                 <span class="vpi-delete local-search-icon" />
@@ -315,25 +318,38 @@ function handleToggleDetail() {
         <template #body>
           <div class="search-dialog" :class="{ 'detail-list': showDetail }">
             <Command.List>
-              <Command.Empty v-if="!searchResult.length">
-                {{ finalSearchConfig?.emptyText || 'No results found.' }}
-              </Command.Empty>
-              <Command.Group v-else :heading="headingText">
-                <Command.Item
-                  v-for="item in showSearchResult" :key="item.route" :data-value="item.route"
-                  @select="handleSelect"
-                >
-                  <div class="link">
-                    <div class="title">
-                      <span class="headings"><i v-if="item.meta.title" class="prefix"># </i>{{ item.meta.title }}</span>
-                      <span v-if="showDateInfo && item.meta.date" class="date">
-                        <!-- @vue-ignore -->
-                        {{ formatShowDateFn(item.meta.date, lang) }}</span>
+              <OverlayScrollbarsComponent :options='{
+                  overflow: {
+                    x: "hidden",
+                  },
+                  scrollbars: {
+                    theme: "os-theme-light",
+                    autoHide: "never",
+                    dragScroll: true,
+                    clickScroll: true,
+                  },
+                }'
+              >
+                <Command.Empty v-if="!searchResult.length">
+                  {{ finalSearchConfig?.emptyText || 'No results found.' }}
+                </Command.Empty>
+                <Command.Group v-else :heading="headingText">
+                  <Command.Item
+                    v-for="item in showSearchResult" :key="item.route" :data-value="item.route"
+                    @select="handleSelect"
+                  >
+                    <div class="link">
+                      <div class="title">
+                        <span class="headings"><i v-if="item.meta.title" class="prefix"># </i>{{ item.meta.title }}</span>
+                        <span v-if="showDateInfo && item.meta.date" class="date">
+                          <!-- @vue-ignore -->
+                          {{ formatShowDateFn(item.meta.date, lang) }}</span>
+                      </div>
+                      <div class="des" v-html="item.meta.description" />
                     </div>
-                    <div class="des" v-html="item.meta.description" />
-                  </div>
-                </Command.Item>
-              </Command.Group>
+                  </Command.Item>
+                </Command.Group>
+              </OverlayScrollbarsComponent>
             </Command.List>
           </div>
         </template>
@@ -394,6 +410,31 @@ function handleToggleDetail() {
   </div>
 </template>
 
+<style lang="css">
+@import 'overlayscrollbars/overlayscrollbars.css';
+.algolia [command-list] {
+  overflow: visible !important;
+  height: auto !important;
+  max-height: none !important;
+}
+.algolia [command-list] [data-overlayscrollbars-contents] {
+  /*height: var(--command-list-height);*/
+  max-height: 360px;
+  overflow: auto;
+  overscroll-behavior: contain;
+  transition: 100ms ease;
+  transition-property: all;
+}
+.algolia [command-list] [data-overlayscrollbars-viewport~="overflowYScroll"] {
+  padding-right: 12px !important;
+}
+@media screen and (max-width: 560px) {
+  .algolia [command-list] [data-overlayscrollbars-contents] {
+    max-height: calc(100vh - 120px);
+  }
+}
+</style>
+
 <style lang="css" scoped>
 .blog-search {
   flex: 1;
diff --git a/src/type.ts b/src/type.ts
index 44edf143c4669e461640aaafc5d30dfd39232505..0bca037b37402e72171374c9652493c578e275be 100644
--- a/src/type.ts
+++ b/src/type.ts
@@ -82,6 +82,21 @@ export interface SearchConfig {
    * 'Search Docs'
    */
   placeholder?: string
+  /**
+   * @default
+   * 'Close search'
+   */
+  closeButtonTitle?: string
+  /**
+   * @default
+   * 'Display detailed list'
+   */
+  displayDetailsButtonTitle?: string
+  /**
+   * @default
+   * 'Reset search'
+   */
+  resetButtonTitle?: string
   /**
    * @default
    * 'No results found.'
