diff --git a/src/Search.vue b/src/Search.vue
index 7e466d132afb6461031cf10ddad1bd5c79da3b2e..71f8daee5968d057f841803dd84de140bcf6fe03 100644
--- a/src/Search.vue
+++ b/src/Search.vue
@@ -11,6 +11,9 @@ import type { SearchConfig, SearchItem } from './type'
 import { extractFuzzyKeywordsFromExcerpts, formatPagefindResult } from './search'
 import { formatShowDate } from './utils'
 
+import VPButton from 'vitepress/dist/client/theme-default/components/VPButton.vue';
+import { OverlayScrollbarsComponent } from 'overlayscrollbars-vue';
+
 // 搜索结果
 const searchResult = ref<Omit<SearchItem, 'result'>[]>([])
 // 是否正在搜索中
@@ -289,9 +292,9 @@ function handleToggleDetail() {
 
 <template>
   <div class="blog-search" data-pagefind-ignore="all">
-    <button class="nav-search-btn-wait" @click="showSearchModal">
+    <VPButton class="nav-search-btn-wait" @click="showSearchModal">
       <span>
-        <svg viewBox="0 0 20 20">
+        <svg width="14" height="14" viewBox="0 0 20 20">
           <path
             d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z"
             stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"
@@ -302,7 +305,7 @@ function handleToggleDetail() {
         finalSearchConfig?.btnPlaceholder || 'Search'
       }}</span>
       <span class="metaKey" />
-    </button>
+    </VPButton>
     <ClientOnly>
       <Command.Dialog :visible="searchModal" theme="algolia" :footer-class="{ 'no-search-result': !searchResult.length }">
         <template #header>
@@ -348,28 +351,41 @@ function handleToggleDetail() {
                   {{ finalSearchConfig?.emptyText || 'No results found.' }}
                 </Command.Empty>
               </template>
-              <Command.Group v-else :heading="headingText">
-                <Command.Item
-                  v-for="item in showSearchResult" :key="item.route" :data-value="item.route"
-                  @select="handleSelect"
-                >
-                  <div class="link">
-                    <div class="title">
-                      <span class="headings">
-                        <i v-if="item.meta.title?.length" class="prefix">#</i>
-                        <template v-for="(heading, i) in item.meta.title" :key="heading">
-                          <i v-if="i" class="vpi-chevron-right local-search-icon" />
-                          <span class="heading" v-html="heading" />
-                        </template>
-                      </span>
-                      <span v-if="showDateInfo && item.meta.date" class="date">
-                        <!-- @vue-ignore -->
-                        {{ formatShowDateFn(item.meta.date, lang) }}</span>
+              <OverlayScrollbarsComponent v-else :options='{
+                  overflow: {
+                    x: "hidden",
+                  },
+                  scrollbars: {
+                    theme: "os-theme-light",
+                    autoHide: "never",
+                    dragScroll: true,
+                    clickScroll: true,
+                  },
+                }'
+              >
+                <Command.Group :heading="headingText">
+                  <Command.Item
+                    v-for="item in showSearchResult" :key="item.route" :data-value="item.route"
+                    @select="handleSelect"
+                  >
+                    <div class="link">
+                      <div class="title">
+                        <span class="headings">
+                          <i v-if="item.meta.title?.length" class="prefix">#</i>
+                          <template v-for="(heading, i) in item.meta.title" :key="heading">
+                            <i v-if="i" class="vpi-chevron-right local-search-icon" />
+                            <span class="heading" v-html="heading" />
+                          </template>
+                        </span>
+                        <span v-if="showDateInfo && item.meta.date" class="date">
+                          <!-- @vue-ignore -->
+                          {{ formatShowDateFn(item.meta.date, lang) }}</span>
+                      </div>
+                      <div class="des" v-html="item.meta.description" />
                     </div>
-                    <div class="des" v-html="item.meta.description" />
-                  </div>
-                </Command.Item>
-              </Command.Group>
+                  </Command.Item>
+                </Command.Group>
+              </OverlayScrollbarsComponent>
             </Command.List>
             <div
               v-if="isSearching && showLoadingMask && searchResult.length"
@@ -430,6 +446,31 @@ function handleToggleDetail() {
   </div>
 </template>
 
+<style lang="css">
+@import 'overlayscrollbars/overlayscrollbars.css';
+.algolia [command-list] {
+  overflow: visible !important;
+  height: auto !important;
+  max-height: none !important;
+}
+.algolia [command-list] [data-overlayscrollbars-contents] {
+  /*height: var(--command-list-height);*/
+  max-height: 360px;
+  overflow: auto;
+  overscroll-behavior: contain;
+  transition: 100ms ease;
+  transition-property: all;
+}
+.algolia [command-list] [data-overlayscrollbars-viewport~="overflowYScroll"] {
+  padding-right: 12px !important;
+}
+@media screen and (max-width: 560px) {
+  .algolia [command-list] [data-overlayscrollbars-contents] {
+    max-height: calc(100vh - 120px);
+  }
+}
+</style>
+
 <style lang="css" scoped>
 .blog-search {
   flex: 1;
diff --git a/src/assets/scss/algolia.scss b/src/assets/scss/algolia.scss
index 933fb0f22c2216849f3fcbc0c48fbba73d79b09a..9766e22446be5e2c1b951e483598ef9c0397ddac 100644
--- a/src/assets/scss/algolia.scss
+++ b/src/assets/scss/algolia.scss
@@ -208,7 +208,7 @@
   box-shadow: none;
 }
 
-// transition for command-dialog
+/* transition for command-dialog
 .algolia {
   &.command-dialog-enter-active,
   &.command-dialog-leave-active {
@@ -231,3 +231,4 @@
     }
   }
 }
+*/
\ No newline at end of file
diff --git a/src/assets/scss/search.css b/src/assets/scss/search.css
index 04e0d04f454b6a6a037106a39c855ecc6dfdddb2..cdf88907e7e931b984454ee1201827815f1de97b 100644
--- a/src/assets/scss/search.css
+++ b/src/assets/scss/search.css
@@ -568,7 +568,7 @@ label.search-icon {
   margin-left: 12px;
 }
 
-.algolia.command-dialog-enter-active [command-dialog-mask],
+/*.algolia.command-dialog-enter-active [command-dialog-mask],
 .algolia.command-dialog-enter-active [command-dialog-wrapper],
 .algolia.command-dialog-leave-active [command-dialog-mask],
 .algolia.command-dialog-leave-active [command-dialog-wrapper] {
@@ -585,7 +585,7 @@ label.search-icon {
 .algolia.command-dialog-leave-to [command-dialog-wrapper] {
   opacity: 0;
   translate: 0 -1rem;
-}
+}*/
 
 .mpa .algolia [command-dialog-mask],
 .mpa .algolia [command-dialog-wrapper] {
